src/hashmap/concurrent_unordered_multimap.cuh

```diff
@@ -412,40 +456,72 @@ public:
                     atomicAdd( &m_collisions, 1 );
                 }
             }
-#else
-#ifdef _OPENMP
-            #pragma omp critical
-#endif
+
+            hash_tbl_idx = (hash_tbl_idx+1)%hashtbl_size;
+
+            attempt_counter++;
+            if( attempt_counter > hashtbl_size)
             {
-                if ( m_equal( unused_key, tmp_it->first ) ) {
-                    hashtbl_values[hash_tbl_idx] = thrust::make_pair( x.first, x.second );
-                    it = tmp_it;
-                }
+              printf("Attempted to insert to multimap but the map is full!\n");
```

Comment:
Currently nothing will happen other than printing the error and exiting the loop, which is better than what would currently happen---infinite loop. 

Technically the hashmap should never be full by construction, but there was a bug where the computation to determine the size of the hash table was overflowing an `int` and therefore was ending up smaller than it was supposed to be. So this is an attempt to at least do something other than cause an infinite loop. We can't really check for `end()` and return an error, because `insert` is called in a device kernel, and we can't return an error code from a kernel.

Reply:
I suggest to have a threshold for the max number of hash checks, and if we exceed that, we store the row index in some preallocated buffer in system memory. Then outside the kernel we print the error, indicating the row number, which could be later extended to passing it properly all the way up to PyGDF. I just don’t like doing a printf from within a kernel

Reply:
How about I pass a `gdf_error *` into the kernel, and if a thread detects that the hashmap is full, then it can set it to `GDF_HASH_TABLE_FULL`. We can then check the status of the `gdf_error` after the build kernel.